File: Binary Search I/java/Day 10/IntersectOfArray.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        int size=nums1.length>nums2.length?nums1.length:nums2.length;
        int[] inter= new int[size];
        int k=0;
        for(int i=0;i<nums1.length;i++){
            for(int j=0;j<nums2.length;j++){
                if(nums1[i]==nums2[j]){
                    inter[k++]=nums1[i];
                    nums2[j]=-1;
                    break;
                }
            }
        }
        int[] ans = new int[k];
        for(int i=0;i<k;i++){
            ans[i]=inter[i];
        } 
        return ans;
    }
}

File: Binary Search I/cpp/Day 10/IntersectOfArray.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        int size=nums1.size()>nums2.size()?nums1.size():nums2.size();
        vector<int> inter;
        int k=0;
        for(int i=0;i<nums1.size();i++){
            for(int j=0;j<nums2.size();j++){
                if(nums1[i]==nums2[j]){
                    inter.push_back(nums1[i]);
                    nums2[j]=-1;
                    break;
                }
            }
        }
        return inter;
    }
};